code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -266,12 +282,52 @@
 		else
 			new_limbs += limb_icon_cache[icon_render_keys[limb.body_zone]] //Pulls existing sprites from the cache
 
+	remove_overlay(BODYPARTS_LAYER)
+
 	if(new_limbs.len)
 		overlays_standing[BODYPARTS_LAYER] = new_limbs
 
 	apply_overlay(BODYPARTS_LAYER)
 
 
+/**
+ * Here we update the legs separately from the other bodyparts. Thanks BYOND for so little support for dir layering.
+ *
+ * Arguments:
+ * * right_leg - Right leg that we might need to update. Can be null.
+ * * left_leg - Left leg that we might need to update. Can be null.
+ * * old_left_leg_key - The icon_key of the left_leg, passed here to avoid having to re-generate it in this proc.
+ *
+ * Returns a boolean, TRUE if the legs need to be redrawn, FALSE if they do not need to be redrawn.
+ * Necessary so that we can ensure that modifications of legs cause overlay updates.
+ */
+/mob/living/carbon/proc/update_legs(obj/item/bodypart/r_leg/right_leg, obj/item/bodypart/l_leg/left_leg, old_left_leg_key)
+	var/list/left_leg_icons // yes it's actually a list, bet you didn't expect that, now did you?
+	var/legs_need_redrawn = FALSE
+	if(left_leg)
+		// We regenerate the look of the left leg if it isn't already cached, we don't if not.
+		if(!(icon_render_keys[left_leg.body_zone] == old_left_leg_key))
+			limb_icon_cache[icon_render_keys[left_leg.body_zone]] = left_leg.get_limb_icon()
+			legs_need_redrawn = TRUE
+
+		left_leg_icons = limb_icon_cache[icon_render_keys[left_leg.body_zone]]
+
+	if(right_leg)
+		var/old_right_leg_key = icon_render_keys?[right_leg.body_zone]
+		var/left_leg_mask_key = left_leg?.generate_mask_key().Join() // We generate a new mask key, to see if it changed.
+		right_leg.left_leg_mask_key = left_leg ? left_leg_mask_key : null
```

Comment:
What is the point of this ternary? ?. will hand back null if the thing it's being called on is null

Reply:
```suggestion
		right_leg.left_leg_mask_key = left_leg?.generate_mask_key().Join() // We generate a new mask key, to see if it changed.
```
could genuinely just do this

Reply:
Also it's weird to call join with no arguments, opens you up to weird caching headaches, might wanna stick a "-" in there

Reply:
That's how it was done elsewhere, I just copied a proc and then removed some of the content.

Reply:
I am responsible for the `.Join()` meme. Key generation was originally a string until I was informed the horror that making a list and using `list.Join()` was faster than string concatenation. Since all of the key parts already were using `"-"`, I just slapped a `.Join()` there.

Reply:
I hate coders

Reply:
I kept it as a separate var to make the following check a little shorter, I know it wasn't necessary, but I did it for readability, mostly.